internal/services/oidc.go

```diff
@@ -106,33 +115,75 @@ func (osvc *OIDCService) IsFlowSecure(state string, token models.Tokens) (bool,
 	return true, nil
 }
 
-// CreateJWT creates a new token and the claims you would like it to contain.
+// CreateJWT creates a new JWT token and Custom Claims.
 func (osvc *OIDCService) CreateJWT(cc *models.CustomClaims) (models.CustomClaims, error) {
-	// For HMAC signing method, the key can be any []byte
-	hmacRandSecret, err := randomByte(1990)
-	if err != nil {
-		osvc.logger.Errorf("An error occurred while generating HMAC. Error: %v", err)
-		return models.CustomClaims{}, err
-	}
+	var err error
 
 	customClaims := models.CustomClaims{
-		Name:    cc.Name,
-		Email:   cc.Email,
-		Profile: cc.Profile,
+		Name:     cc.Name,
+		Email:    cc.Email,
+		Profile:  cc.Profile,
+		Audience: cc.Audience,
 		RegisteredClaims: jwt.RegisteredClaims{
-			ExpiresAt: jwt.NewNumericDate(time.Now().Add(time.Duration(osvc.client.Configuration.JWTExpires))),
+			ExpiresAt: jwt.NewNumericDate(time.Now().Add(time.Duration(osvc.configuration.UserJWTExpires))),
 			IssuedAt:  jwt.NewNumericDate(time.Now()),
 		},
 	}
 
 	token := jwt.NewWithClaims(jwt.SigningMethodHS256, customClaims)
 
 	// Sign and get the complete encoded token as a string using the secret.
-	customClaims.AccessToken, err = token.SignedString(hmacRandSecret)
+	customClaims.AccessToken, err = token.SignedString(hmacSecret)
 
 	return customClaims, err
 }
 
+// ValidateJWT validates the token and returns the claims.
+// even if there is an error, return the invalid token instead of nil.
+func (osvc *OIDCService) ValidateJWT(tokenString string) (map[string]interface{}, error) {
```

Comment:
This method `ValidateJWT` does more than a simple validation, it also wrap the token from provider. Let's think in another name or split that responsibilities.

`tokenString`  isn't a good name for this parameter, it is is a token from oidc provider.

Reply:
no, this is already our JWT 🤔

Reply:
This method ValidateJWT does more than validate.